commands/moderation.js

```diff
@@ -1,165 +1,273 @@
 const GReYBot = require('../greybot');
 
 exports.commands = [
-	'prune',
+	'ban',
 	'kick',
-	'ban'
+	'prune'
 ]
 
 var lastPruned = new Date().getTime() - (GReYBot.Config.pruneInterval * 1000);
 
-function resolveMention(usertxt){
+function resolveMention(usertxt) {
 	var userid = usertxt;
-	if(usertxt.startsWith('<@!')){
-		userid = usertxt.substr(3,usertxt.length-4);
+	if (usertxt.startsWith('<@!')) {
+		userid = usertxt.substr(3, usertxt.length - 4);
 	} else {
-		if(usertxt.startsWith('<@')){
-			userid = usertxt.substr(2,usertxt.length-3);
+		if (usertxt.startsWith('<@')) {
+			userid = usertxt.substr(2, usertxt.length - 3);
 		}
 	}
 	return userid;
 }
 
-exports.prune = {
-	usage: '<count|1-100>',
-	process: (msg, suffix) => {
-		var count = parseInt(suffix) || 0;
-
-		if (msg.member.hasPermission('MANAGE_MESSAGES', true)) {
-			var timeSinceLastPrune = Math.floor(new Date().getTime() - lastPruned);
-
-			if (timeSinceLastPrune > (GReYBot.Config.pruneInterval * 1000)) {
-				count++;
-				if (count > 1) {
-					if (count > GReYBot.Config.pruneMax) count = GReYBot.Config.pruneMax;
-
-					msg.channel.fetchMessages({ limit: count })
-						.then(messages => messages.map(m => m.delete()))
-						.then(function () {
-							msg.channel.send({
-								embed: {
-									color: GReYBot.Config.defaultEmbedColor,
-									description: `Pruning ${count - 1} messages...`
-								}
-							}).then(message => message.delete(5000));
-						});
+exports.ban = {
+	usage: '<user> [days of messages to delete] [reason]',
+	description: 'bans the user, optionally deleting messages from them in the last x days',
+	process: function (msg, suffix) {
+		var args = suffix.split(' ');
+		if (args.length > 0 && args[0]) {
+			if (!msg.guild.me.hasPermission('BAN_MEMBERS', true) || !msg.guild.me.hasPermission('ADMINISTRATOR')) {
```

Comment:
this is not necessary. looking at the api you can see that .hasPermission accepts an "admin override" boolean. so the double-check is redundant.